src/filmgrain.rs

```diff
@@ -133,9 +136,14 @@ impl fgy_32x32xn::Fn {
         let bh = bh as c_int;
         let row_num = row_num as c_int;
         let bd = bd.into_c();
-        self.get()(
-            dst_row, src_row, stride, data, pw, scaling, grain_lut, bh, row_num, bd,
-        )
+        let dst = FFISafe::new(dst);
+        let src = FFISafe::new(src);
+        // SAFETY: Fallback `fn fgy_32x32xn_rust` is safe; asm is supposed to do the same.
+        unsafe {
+            self.get()(
+                dst_row, src_row, stride, data, pw, scaling, grain_lut, bh, row_num, bd, dst, src,
```

Comment:
I really don't like the idea of adding extra arguments to ASM calls. This will add unnecessary stack manipulation to add the params before the call. Please be careful with this and benchmark this change.

Reply:
I haven't noticed any changes in the benchmarks.  I just doubt an extra `push` or `str` or two is going to have noticeable perf effects.

Reply:
Ok, just wanted to highlight this as a thing worth checking.

Reply:
Okay, I'll be sure to check it each time I add an extra arg.